File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Sleep.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.LuaInterface;

@LuaInterface


public class Sleep extends FlavourBuff {
	
	public static final float SWS	= 1.5f;

	@Override
	public void attachVisual() {
		target.getSprite().idle();
	}
}

File: scripts/buffs/Sleep.lua
---
--- Sleep: magical sleep, AI dispatch handled in Mob.add (was actors/buffs/Sleep.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

return buff.init{
    desc  = function ()
        return {
            icon          = -1, -- no indicator icon
            name          = "SleepBuff_Name",
            info          = "SleepBuff_Info",
        }
    end,

    attachVisual = function(self, buff)
        buff.target:getSprite():idle()
    end
}
